src/metabase/pulse.clj

```diff
@@ -65,21 +65,36 @@
                               (str "Pulse: " (:name pulse))
                               attachments)))
 
+(defn- is-card-empty?
+  "Check if the card is empty"
+  [card]
+  (let [result (:result card)]
+    (or (zero? (-> result :row_count))
+        ;; Many aggregations result in [[nil]] if there are no rows to aggregate after filters
+        (= [[nil]]
+           (-> result :data :rows)))))
+
+(defn are-all-cards-empty?
```

Comment:
It looks like this function is only used in this namespace, so you should probably make it private `defn-`